src/main/resources/templates/db_scripts/oaipmh/createOaiPmhViewFunction.sql

```diff
@@ -0,0 +1,190 @@
+    returns table
+            (
+                instanceId          uuid,
+                instanceUpdatedDate timestamp with time zone,
+                deleted             boolean,
+                instanceFields      jsonb
+            )
+as
+$body$
+with instanceIdsInRange as (
+    select inst.id                                              as instanceId,
+           (inst.jsonb #>> '{metadata,updatedDate}')::timestamp as maxDate
+    FROM ${myuniversity}_${mymodule}.instance inst
+    where (inst.jsonb #>> '{metadata,updatedDate}')::timestamp between dateOrMin($1) and dateOrMax($2)
+
+    union all
+    select instanceid,
+           greatest((item.jsonb #>> '{metadata,updatedDate}')::timestamp,
+                    (hr.jsonb #>> '{metadata,updatedDate}')::timestamp) as maxDate
+    from holdings_record hr
+             join ${myuniversity}_${mymodule}.item item
+                  on item.holdingsrecordid = hr.id
+    where ((hr.jsonb #>> '{metadata,updatedDate}')::timestamp between dateOrMin($1) and dateOrMax($2)
+        or
+           (item.jsonb #>> '{metadata,updatedDate}')::timestamp between dateOrMin($1) and dateOrMax($2))
+
+    union all
+    select (audit_holdings_record.jsonb #>> '{record,instanceId}')::uuid,
+           greatest(
+                   (audit_item.jsonb #>> '{record,updatedDate}')::timestamp,
+                   (audit_holdings_record.jsonb #>>
+                    '{record,updatedDate}')::timestamp) as maxDate
+    from audit_holdings_record audit_holdings_record
+             join audit_item audit_item
+                  on (audit_item.jsonb ->> '{record,holdingsRecordId}')::uuid =
+                     audit_holdings_record.id
+    where ((audit_holdings_record.jsonb #>>
+            '{record,updatedDate}')::timestamp between dateOrMin($1) and dateOrMax($2)
+        or
+           (audit_item.jsonb #>> '{record,updatedDate}')::timestamp between dateOrMin($1) and dateOrMax($2))
+),
+     instanceIdsAndDatesInRange as (SELECT instanceId,
+                                           max(instanceIdsInRange.maxDate) as maxDate
+                                    from instanceIdsInRange,
+                                         instance
+
+                                    where instanceIdsInRange.maxDate between dateOrMin($1) and dateOrMax($2)
+                                      and instance.id = instanceIdsInRange.instanceId
+                                      and not (
+                                            $4 and
+                                            coalesce(
+                                                    (instance.jsonb ->> 'discoverySuppress')::bool,
+                                                    false))
+                                    group by 1
+     )
+
+select instanceIdsAndDatesInRange.instanceId,
+       instanceIdsAndDatesInRange.maxDate,
+       false as deleted,
+       (select to_jsonb(itemAndHoldingsAttrs) as instanceFields
```

Comment:
is instanceFields a misleading name here?  Doesn't this include data from location/holding/item records?